content/articles/image-watermarking-with-opencv-python/index.md

```diff
@@ -0,0 +1,155 @@
+
+```python
+import cv2
+import numpy as np
+import glob
+import os
+
+from numpy._distributor_init import filename
+
+logo = cv2.imread("elite_logo.jpg") #loading logo image
+h_logo, w_logo, _ = logo.shape #getting height and width
+```
+![Logo Loaded](/engineering-education/image-watermarking-with-opencv-python/logoloaded.jpg)
+#### Getting Images to Be Watermarked
+Once we have the logo we get the path of the images that are to be watermarked, this is done using the `glob library` that is already imported above. When the images are located a loop is created to load all the images getting their heights and widths. The images are located in the `flowers` directory as shown.
+![Image Folder](/engineering-education/image-watermarking-with-opencv-python/flowers.jpg).
+
+```python
+images_path = glob.glob("flowers/*.*") #flowers is the folder holding the flower images to be watermarked.
+print("Adding watermark")
+for img_path in images_path: #loop loading the images and getting width and height
+    img = cv2.imread(img_path)
+    h_img, w_img, _ = img.shape
+```
+#### Placing Watermark
+The location of the watermark depends on the user. It is possible to place the location anywhere, but this article will place the watermark at the center of the images. Will write a code to get the center of all the images and use the `cv2.addWeighted()`to specify the opacity of the logo for it to look nice and smooth when placed over the images.
+```python
+#get the image's center and the spot where the watermark should be placed
+center_y = int(h_img / 2)
+center_x = int(w_img / 2)
+top_y = center_y - int(h_logo / 2)
+left_x = center_x - int(w_logo / 2)
+bottom_y = top_y + h_logo
+right_x = left_x + w_logo
+```
+Having the center and location we specify the **Region of Interest(ROI)** on the images where the watermark will be placed then add the logo mark to the images;
+```python
+roi = img[top_y:bottom_y, left_x:right_x] #getting region of interest
+
+result = cv2.addWeighted(roi, 1, logo, 0.3, 0) #adding logo to the ROI
+img[top_y:bottom_y, left_x:right_x] = result #replacing ROI on the image
+```
+The output sample of a watermarked flower image is as shown below;
+Original image.
+![Original Flower Image](/engineering-education/image-watermarking-with-opencv-python/flowerB.jpg).
+Watermarked image.
+![Watermarked Flower Image](/engineering-education/image-watermarking-with-opencv-python/watermarked_flowerB.jpg).
+### Saving the Watermarked Images
+After the process of watermarking we get the filenames of the images and save the watermarked images on a different folder, `watermarked_images` folder as shown;
+```python
+filename= os.path.basename(img_path)
+
+cv2.imwrite("watermarked_images/watermarked_" + filename, img) #watermarked_images is the folder holding the watermarked images.
+
+print("Watermark added to all Images")
+```
+When the whole code is combined and run successfully it displays the output of the two print functions on the command line.
```

Comment:
Please show us using code how the combined code looks like before showing us the output.